src/DataGrid.tsx

```diff
@@ -940,32 +950,41 @@ function DataGrid<R, SR, K extends Key>(
     return isDraggedOver ? selectedPosition.idx : undefined;
   }
 
-  function getSelectedCellProps(rowIdx: number): SelectedCellProps | EditCellProps<R> | undefined {
+  function getSelectedCellProps(rowIdx: number): SelectedCellProps<R, SR> | undefined {
     if (selectedPosition.rowIdx !== rowIdx) return;
 
     if (selectedPosition.mode === 'EDIT') {
+      const row = rows[selectedPosition.rowIdx];
+      if (isGroupRow(row)) return;
+      const column = columns[selectedPosition.idx];
+      const colSpan = getColSpan(column, lastFrozenColumnIndex, { type: 'ROW', row });
```

Comment:
May be we can use `row: selectedPosition.row`

Reply:
Worth a try